lib/stringify.js

```diff
@@ -43,9 +48,9 @@ var stringify = function stringify(object, prefix, generateArrayPrefix, strictNu
 
     if (typeof obj === 'string' || typeof obj === 'number' || typeof obj === 'boolean' || utils.isBuffer(obj)) {
         if (encoder) {
-            return [encoder(prefix) + '=' + encoder(obj)];
+            return [formatter(encoder(prefix)) + '=' + formatter(encoder(obj))];
         }
-        return [prefix + '=' + String(obj)];
+        return [formatter(prefix) + '=' + formatter(String(obj))];
```

Comment:
I don't see how it can be formatted differently if user provide only one option. Formatter is dedicated function for string variables - keys and values and "=" sign and concatenation is not an formatter responsibility.